python/pyspark/sql/readwriter.py

```diff
@@ -446,6 +450,9 @@ def csv(self, path, schema=None, sep=None, encoding=None, quote=None, escape=Non
                               If None is set, it uses the default value, ``1.0``.
         :param emptyValue: sets the string representation of an empty value. If None is set, it uses
                            the default value, empty string.
+        :param locale: sets a locale as language tag in IETF BCP 47 format. If None is set,
+                       it uses the default value, ``en-US``. For instance, ``locale`` is used while
+                       parsing dates and timestamps.
```

Comment:
It seems parsing decimals using `locale` will be slightly tricky in JSON case because we leave this to Jackson by calling its method `getCurrentToken` and `getDecimalValue`, and I haven't found how to pass locale to it. Probably we will need a custom deserialiser?

In the CSV case, it should be easier since we convert strings ourselves. I will try to do that for CSV first of all when this PR be merged.

Reply:
Here is the PR for parsing decimals from CSV: https://github.com/apache/spark/pull/22979